### PR TITLE
Add support for ApplicationGatewayWebApplicationFirewallPolicies

### LINKED ISSUE #4119: Feature: Support Microsoft.Network/ApplicationGatewayWebApplicationFirewallPolicies
The Swagger for this resource is [here](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/network/resource-manager/Microsoft.Network/stable/2024-01-01/webapplicationfirewall.json#L295).

It's used like [this](https://github.com/Azure/azure-quickstart-templates/blob/master/quickstarts/microsoft.network/application-gateway-waf-firewall-policy/azuredeploy.json#L613C125-L613C173) and we already support the Network.FrontDoor equivalent `​Microsoft.Network/​FrontDoorWebApplicationFirewallPolicies` which is located [here](https://azure.github.io/azure-service-operator/reference/network.frontdoor/v1api20220501/#network.frontdoor.azure.com/v1api20220501.WebApplicationFirewallPolicy)

### PR DESCRIPTION
**Pull request by matthchr**
Closes #4119.

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests
- [x] this PR contains YAML Samples

### CODE REVIEW

File: v2/api/network/customizations/bastion_host_extension_types.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	"github.com/go-logr/logr"
+
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/core"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/extensions"
+)
+
+var _ extensions.ErrorClassifier = &BastionHostExtension{}
+
+func (extension *BastionHostExtension) ClassifyError(
```
Comment (matthchr) [inline v2/api/network/customizations/bastion_host_extension_types.go:15]: Found these during recording, so just fixed inline.

File: v2/samples/network/v1api20220701/v1api20240101_webapplicationfirewallpolicy.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+apiVersion: network.azure.com/v1api20240101
+kind: WebApplicationFirewallPolicy
```
Comment (matthchr) [inline v2/samples/network/v1api20220701/v1api20240101_webapplicationfirewallpolicy.yaml:2]: This is a weird place for this resource sample, my plan is to create new versions of all of the networking resources and then I'll move this sample to a 2024 folder

Comment (matthchr) [reply v2/samples/network/v1api20220701/v1api20240101_webapplicationfirewallpolicy.yaml:2]: Will do that in a separate PR though

**Review by theunrepentantgeek — APPROVED** [review]

**Pull request merged**